### PR TITLE
Tutorial progress indicator feature

### PR DESCRIPTION
**Pull request by eojthebrave**
Tutorial progress indicator front-end. Still requires updates to connect with Drupal API once that's ready.

Adds two new custom hooks:

`useTutorialList()`, and `useProgressIndicator()` that later makes use of the former. The extra abstraction allows for using the same list of cached tutorials to handle additional peronsonalization data in the future if we need to.

I also ended up opting to use `localStorage` for the cached list for now as it's easier, and doesn't require as much code as `IndexedDB`. I did create a `LocalCache` class as a wrapper around localStorage which will make it easier to switch to IndexedDB in the future if we need to because we can write another implementation of the same class/interface.

I also added/updated a bunch of tests:

- I made the DrupalOauthProvider tests a bit more useful in that they actually test to prove that when a user changes from anon to authenticated the state is updated properly. Which is the most important thing this context provides. I had to refactor these anyways because I added some code to the provider that handles exchanging an OAuth token for a Drupal user's UUID.
- I added a bunch of tests for the new `useProgressIndicator()` hook to ensure it handles things like loading data, updating and purging the cache, etc. This, in a round about way, also tests all the essential features of the `useTutorialList` hook, so I did not provide additional tests there ...
- ... except for one that ensures that the localStorage cache is purged when a user changes from anon to authenticated or visa versa as that logic was tricky to figure out how to implement and therefore I'm probably likely to accidentally break it in the future. :)
- Updates to various other tests that rely on the OAuth provider to ensure they still pass

## Changes!

I bumped this module's version to `1.1.0` because it contains changes that are sort of breaking, and implements a new API.

The _kind of_ breaking change is that the gatsby-browser.js now does all the work of setting up the `DrupalOauth` client and wrapping the whole application in `DrupalOauthContext.Provider` so that no longer needs to be done by individual applications using this theme. Right now for both HN and GG that's happening in the layout.jsx file. And that code can be removed when updating to this version of the theme.

### CODE REVIEW

**Comment by eojthebrave** [comment]
This still needs to have code added for POSTing updates to the Drupal API once we have an API that allows us to do that.

File: src/components/Tutorial/TutorialProgressIndicator.jsx [hunk 1]
```diff
@@ -9,7 +9,15 @@ const TutorialProgressIndicator = ({
   currentUserId,
   ...rest
 }) => {
-  const [progress, markAsRead, markAsUnread] = useProgressIndicator(complete, entityId);
+  // If the user is no logged in there's nothing left to do.
```
Comment (blakehall) [inline src/components/Tutorial/TutorialProgressIndicator.jsx]: s/no/not

**Review by blakehall — APPROVED** [review]
This looks 👍 to me (out of full context), with one small typo to fix.

I'll work on testing this (in context) on my local here shortly.

**Comment by blakehall** [comment]
This is working, and ready to merge.

**Pull request merged**